Data-Structures/src/main/java/linkedlist/LinkedList.java

```diff
@@ -31,5 +31,45 @@ public String toString(){
     return allTheThings;
   }
 
+  public void append(T value){
+    if(head == null){
+      head = new Node(value, null);
+      return;
+    }
+    this.current = head;
+    while(this.current.next != null){
+      this.current = current.next;
+    }
+    Node <T> newNode = new Node(value, null);
+    current.next = newNode;
+  }
+
+  public void insertBefore(T value, T target){
+    if(head == null){
+      head = new Node(value, null);
+    }
+    this.current = head;
+    while(current.next != null){
+      if(current.value == target){
+        Node newNode = new Node(value, current);
```

Comment:
These two lines are creating an infinite loop. newNode points to current and current points to newNode